pkg/discovery/dtofactory/container_spec_entity_dto_builder.go

```diff
@@ -133,3 +147,42 @@ func (builder *containerSpecDTOBuilder) getCommoditiesSold(containerSpecMetrics
 	}
 	return commoditiesSold, nil
 }
+
+func aggregateThrottlingSamples(samples interface{}) (float64, float64, float64, bool) {
+	typedSamples, ok := samples.([][]metrics.ThrottlingCumulative)
+	if !ok {
+		return 0, 0, 0, false
+	}
+
+	var throttledOverall, totalOverall, peakOverall float64
+	for _, singleContainerSamples := range typedSamples {
+		numberOfSamples := len(singleContainerSamples)
+		if numberOfSamples <= 1 {
+			// We don't have enough samples to calculate this value.
+			continue
+		}
+		sort.SliceStable(singleContainerSamples, func(i, j int) bool {
+			return singleContainerSamples[i].Timestamp < singleContainerSamples[j].Timestamp
+		})
```

Comment:
Yes, this actually happens on a full discovery, where the full discovery data is appended at the head of the list rather then at the end. I checked altering the merge of sinks before introducing this sort, but that change was much more intrusive and I did not want to break things. 
On later thoughts, I would want to keep this sort even if we merge sinks in order because: 
1. The performance penalty is almost nil on an almost sorted list (as is currently the case, ie. one sample out of place).
2. This also makes this portion of code independent from how we merge the sinks.